test/box-luatest/gh_6246_mvcc_different_select_behavior_test.lua

```diff
@@ -0,0 +1,103 @@
+local server = require('test.luatest_helpers.server')
+local t = require('luatest')
+
+local g = t.group()
+
+g.before_all = function()
+    g.server = server:new{
+        alias   = 'default',
+        box_cfg = {memtx_use_mvcc_engine = true}
+    }
+    g.server:start()
+end
+
+g.after_all = function()
+    g.server:drop()
+end
+
+g.test_mvcc_different_select_behavior = function()
+    g.server:exec(function()
+        local t = require('luatest')
+        local fiber = require('fiber')
+
+        local s = box.schema.space.create('test')
+        s:create_index('primary')
+
+        local f = fiber.create(function()
+            fiber.self():set_joinable(true)
+            s:insert{1}
+        end)
+
```

Comment:
Your test if flaky: adding `fiber.sleep` here breaks it. Please consider using `ERRINJ_WAL_DELAY`

Reply:
Upd: I guess I am wrong, and your test are correct: the transaction in fiber `f` does not get committed until `fiber.yield` — if so, please ignore all other comments about `ERRINJ_WAL_DELAY`.